pkg/disttask/framework/taskexecutor/task_executor.go

```diff
@@ -166,6 +166,10 @@ func (s *BaseTaskExecutor) Run(ctx context.Context, task *proto.Task) (err error
 	if s.mu.handled {
 		return err
 	}
+	if err == nil {
+		// may have error in defer function in run(ctx, task)
+		err = s.getError()
```

Comment:
we depends on too much on `getError()`, it's actually replacing the responsiblity of `error of method return`

but ok for now

Reply:
Will refactor it